docs/calling-downstream-apis/README.md

```diff
@@ -1,22 +1,431 @@
-# Calling Downstream APIs
+# Calling Downstream APIs with Microsoft.Identity.Web
+
+This guide helps you choose and implement the right approach for calling downstream APIs (Microsoft Graph, Azure services, or custom APIs) from your ASP.NET Core or OWIN applications using Microsoft.Identity.Web.
+
+## 🎯 Choosing the Right Approach
+
+Use this decision tree to select the best method for your scenario:
 
-## Decision Tree
 ```mermaid
-graph TD;
-    A[Use Graph?] -->|Yes| B[Use Microsoft Graph SDK];
-    A -->|No| C[Use Azure SDK?];
-    C -->|Yes| D[Use Azure SDK Client];
-    C -->|No| E[Use IDownstreamApi?];
-    E -->|Yes| F[Use IDownstreamApi];
-    E -->|No| G[Use MicrosoftIdentityMessageHandler?];
-    G -->|Yes| H[Use MicrosoftIdentityMessageHandler];
-    G -->|No| I[Use IAuthorizationHeaderProvider];
-    I -->|Yes| J[Use IAuthorizationHeaderProvider];
-    I -->|No| K[Handle Errors];
-
-```
-
-## Token Acquisition Patterns
-- Overview of token acquisition patterns
-- Error handling strategies
-- Cross-references to related documents
\ No newline at end of file
+graph TD
+    A[What API are you calling?] --> B{Microsoft Graph?}
+    B -->|Yes| C[Use GraphServiceClient]
+    B -->|No| D{Azure SDK<br/>Storage, KeyVault, etc?}
+    D -->|Yes| E[Use MicrosoftIdentityTokenCredential<br/>with Azure SDK clients]
+    D -->|No| F{Custom API or<br/>other service?}
+    F -->|Yes| G{Need flexible<br/>HTTP control?}
+    G -->|No| H[Use IDownstreamApi<br/>Simple, configured approach]
+    G -->|Yes| I{Using HttpClient<br/>with DI?}
+    I -->|Yes| J[Use MicrosoftIdentityMessageHandler<br/>Composable HTTP pipeline]
+    I -->|No| K[Use IAuthorizationHeaderProvider<br/>Maximum flexibility]
+    
+    style C fill:#d4edda
+    style E fill:#d4edda
+    style H fill:#d4edda
+    style J fill:#d4edda
+    style K fill:#d4edda
+```
+
+## 📊 Comparison Table
+
+| Approach | Best For | Complexity | Configuration | Flexibility |
+|----------|----------|------------|---------------|-------------|
+| **GraphServiceClient** | Microsoft Graph APIs | Low | Simple | Medium |
+| **MicrosoftIdentityTokenCredential** | Azure SDK clients | Low | Simple | Low |
+| **IDownstreamApi** | REST APIs with standard patterns | Low | JSON + Code | Medium |
+| **MicrosoftIdentityMessageHandler** | HttpClient with auth pipeline | Medium | Code | High |
+| **IAuthorizationHeaderProvider** | Custom auth logic | High | Code | Very High |
+
+## 🔐 Token Acquisition Patterns
+
+Microsoft.Identity.Web supports three main token acquisition patterns:
+
+```mermaid
+graph LR
```

Comment:
Let's change this, by:
- Web app calling web APIs on behalf of users
- Web APIs calling web APIs on behalf of users (OBO)
- Web apps, web APIs and daemon calling web APIs on their own behalf
- Daemon scenarios with a user identity (Agent user identities)